src/Entity/Product.php

```diff
@@ -19,11 +21,15 @@ class Product implements \App\Service\Catalog\Product
     #[ORM\Column(type: 'integer', nullable: false)]
     private string $priceAmount;
 
-    public function __construct(string $id, string $name, int $price)
+    #[ORM\Column(type: 'datetime_immutable', nullable: false)]
+    private ?DateTimeImmutable $createdAt;
+
+    public function __construct(string $id, string $name, int $price, ?DateTimeImmutable $createdAt = null)
     {
-        $this->id = Uuid::fromString($id);
-        $this->name = $name;
+        $this->id          = Uuid::fromString($id);
+        $this->name        = $name;
         $this->priceAmount = $price;
+        $this->createdAt   = $createdAt;
```

Comment:
As we already altering DB, maybe it is also worth to add status property here(bool active/inactive) so it can be used to provide more flexibility in the system? Of course this will introduce more complexity to the existing logic...